app/models/world_location_email_signup.rb

```diff
@@ -0,0 +1,80 @@
+class WorldLocationEmailSignup
+  attr_reader :feed
+
+  def initialize(feed)
+    @feed = feed
+  end
+
+  def save
+    return false unless valid?
+
+    ensure_subscriber_list_exists
+    true
+  end
+
+  def signup_url
+    ensure_subscriber_list_exists["subscriber_list"]["subscription_url"]
+  end
+
+  def topic_id
+    ensure_subscriber_list_exists["subscriber_list"]["gov_delivery_id"]
+  end
+
+  def name
+    world_location.name
+  end
+
+  def valid?
+    uri && recognised_url? && world_feed_type? && world_location.present?
+  end
+
+private
+
+  def ensure_subscriber_list_exists
+    @ensure_subscriber_list_exists ||= Services.email_alert_api.find_or_create_subscriber_list(criteria)
```

Comment:
This looks pretty whacky. I know you didn't write it originally

I'd suggest you change this method to be called subscriber_list

and then change the save method to be:

```
def save
    return unless valid?

    subscriber_list
  end
```

not too sure why it was previously set to booleans as it doesn't really need them

Reply:
True, I've gone one step further and removed some of these methods as they aren't actually needed.